pkg/streamingpromql/planning/core/deduplicate_and_merge.go

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package core
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/prometheus/prometheus/promql/parser"
+	"github.com/prometheus/prometheus/promql/parser/posrange"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/operators"
+	"github.com/grafana/mimir/pkg/streamingpromql/planning"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+)
+
+type DeduplicateAndMerge struct {
+	*DeduplicateAndMergeDetails
+	Inner planning.Node
+}
+
+func (d *DeduplicateAndMerge) Details() proto.Message {
+	return d.DeduplicateAndMergeDetails
+}
+
+func (d *DeduplicateAndMerge) NodeType() planning.NodeType {
+	return planning.NODE_TYPE_DEDUPLICATE_AND_MERGE
+}
+
+func (d *DeduplicateAndMerge) Children() []planning.Node {
+	return []planning.Node{d.Inner}
+}
+
+func (d *DeduplicateAndMerge) SetChildren(children []planning.Node) error {
+	if len(children) != 1 {
+		return fmt.Errorf("node of type DeduplicateAndMerge supports 1 child, but got %d", len(children))
+	}
+
+	d.Inner = children[0]
+
+	return nil
+}
+
+func (d *DeduplicateAndMerge) EquivalentTo(other planning.Node) bool {
+	otherDedup, ok := other.(*DeduplicateAndMerge)
+
+	return ok && d.Inner.EquivalentTo(otherDedup.Inner)
+}
+
+func (d *DeduplicateAndMerge) Describe() string {
+	return "deduplicate and merge"
```

Comment:
This could be an empty string, the name of the node is enough to explain what this node does.

(contrast this with an aggregation node, where it's useful to know if it's a sum or an average etc.)